src/libsync/propagateupload.h

```diff
@@ -176,49 +258,62 @@ class PropagateUploadFileQNAM : public PropagateItemJob {
     int _currentChunk;
     int _chunkCount; /// Total number of chunks for this file
     int _transferId; /// transfer id (part of the url)
-    QElapsedTimer _duration;
-    QVector<AbstractNetworkJob*> _jobs; /// network jobs that are currently in transit
-    bool _finished; // Tells that all the jobs have been finished
 
-    // measure the performance of checksum calc and upload
-    Utility::StopWatch _stopWatch;
-
-    QByteArray _transmissionChecksum;
-    QByteArray _transmissionChecksumType;
+    quint64 chunkSize() const { return _propagator->chunkSize(); }
 
-    bool _deleteExisting;
 
-    quint64 chunkSize() const { return _propagator->chunkSize(); }
 
 public:
-    PropagateUploadFileQNAM(OwncloudPropagator* propagator,const SyncFileItemPtr& item)
-        : PropagateItemJob(propagator, item), _startChunk(0), _currentChunk(0), _chunkCount(0), _transferId(0), _finished(false), _deleteExisting(false) {}
-    void start() Q_DECL_OVERRIDE;
+    PropagateUploadFileV1(OwncloudPropagator* propagator,const SyncFileItemPtr& item) :
+        PropagateUploadFileCommon(propagator,item) {}
 
-    /**
-     * Whether an existing entity with the same name may be deleted before
-     * the upload.
-     *
-     * Default: false.
-     */
-    void setDeleteExisting(bool enabled);
+    void doStartUpload() Q_DECL_OVERRIDE;
 
 private slots:
+    void startNextChunk();
     void slotPutFinished();
-    void slotPollFinished();
     void slotUploadProgress(qint64,qint64);
-    void abort() Q_DECL_OVERRIDE;
-    void startNextChunk();
-    void finalize(const SyncFileItem&);
-    void slotJobDestroyed(QObject *job);
-    void slotStartUpload(const QByteArray& transmissionChecksumType, const QByteArray& transmissionChecksum);
-    void slotComputeTransmissionChecksum(const QByteArray& contentChecksumType, const QByteArray& contentChecksum);
-    void slotComputeContentChecksum();
+};
 
+/**
+ * @ingroup libsync
+ *
+ * Propagation job, impementing the new chunking agorithm
+ *
+ */
+class PropagateUploadFileNG : public PropagateUploadFileCommon {
+    Q_OBJECT
 private:
-    void startPollJob(const QString& path);
-    void abortWithError(SyncFileItem::Status status, const QString &error);
+    quint64 _sent; /// amount of data that was already sent
+    uint _transferId; /// transfer id (part of the url)
+    int _currentChunk; /// Id of the next chunk that will be sent
+    QMap<int, quint64> _serverChunks; // Map chunk number with its size  from the PROPFIND on resume
+
+    quint64 chunkSize() const { return _propagator->chunkSize(); }
+    /**
+     * Return the URL of a chunk.
+     * If chunk == -1, returns the URL of the parent folder containing the chunks
```

Comment:
yeah... That's private API anyway.
Having another function mean either duplicating code, or call UrlConcat twice.